### PR TITLE
Remove alpha APIs, Add v1beta2 APIs

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
- Remove alpha APIs
- Add v1beta2 APIs

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Remove alpha APIs, Add v1beta2 APIs
```

### CODE REVIEW

File: config/crd/patches/webhook_in_ibmpowervsclustertemplates.yaml [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-    webhookClientConfig:
-      # this is "\n" used as a placeholder, otherwise it will be rejected by the apiserver for being blank,
-      # but we're going to set it later using the cert-manager (or potentially a patch if not using cert-manager)
-      caBundle: Cg==
-      service:
-        namespace: system
-        name: webhook-service
-        path: /convert
+    webhook:
+      conversionReviewVersions: ["v1", "v1beta1"]
```
Comment (mkumatag) [inline config/crd/patches/webhook_in_ibmpowervsclustertemplates.yaml:11]: why is this `["v1", "v1beta1"]`? why not `["v1beta1", "v1beta2"]`

File: config/samples/cluster.yaml [hunk 2]
```diff
@@ -1,39 +0,0 @@
----
```
Comment (mkumatag) [inline config/samples/cluster.yaml:1]: All the samples are removed, any specific reason for that?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: api/v1beta2/ibmpowervsmachine_types.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	capiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply api/v1beta2/ibmpowervsmachine_types.go:23]: v1beta2 APIs for CAPI not added yet, I hope my understanding is correct here.

File: config/samples/cluster.yaml (hunk 2 above)
Comment (mkumatag) [reply config/samples/cluster.yaml:1]: sgtm

File: api/v1beta2/ibmvpccluster_types.go [hunk 4]
```diff
@@ -14,15 +14,14 @@ See the License for the specific language governing permissions and
...
 */
 
-package v1alpha4
+package v1beta2
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	capiv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	capiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
```
[the remark this answers is not in the document]
Comment (mkumatag) [reply api/v1beta2/ibmvpccluster_types.go:22]: oops, this was mistake! I somehow was under the impression that this is pointing to capibm api version, sorry..

File: api/v1beta2/ibmpowervsmachine_types.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mkumatag) [reply api/v1beta2/ibmpowervsmachine_types.go:23]: oops, this was mistake! I somehow was under the impression that this is pointing to capibm api version, sorry..

File: metadata.yaml [hunk 5]
```diff
@@ -17,3 +14,6 @@ releaseSeries:
   - major: 0
     minor: 4
     contract: v1beta1
+  - major: 0
+    minor: 5
+    contract: v1beta2
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply metadata.yaml]: Fixed.

File: config/crd/patches/webhook_in_ibmpowervsclustertemplates.yaml (hunk 1 above)
Comment (Prajyot-Parab) [reply config/crd/patches/webhook_in_ibmpowervsclustertemplates.yaml:11]: will spend some more time on this, fix as part of a separate PR

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**